src/slack_block_util.ts

```diff
@@ -333,6 +354,170 @@ export function getErrorSlackView(
   };
 }
 
+interface OpenCloseModalProps {
+  /** Selected state or region */
+  stateOrRegionName?: string;
+  /** Selected filter type, if any */
+  channelType?: PodUtil.CHANNEL_TYPE;
+  /** Channels to display, if any */
+  channelRows?: PodUtil.ChannelInfo[];
+}
+
+const getOptionForStateOrRegion = (stateOrRegionName: string): SlackOption => ({
+  text: {
+    type: 'plain_text',
+    text: stateOrRegionName,
+  },
+  value: stateOrRegionName,
+});
+
+const getOptionForChannelType = (value: PodUtil.CHANNEL_TYPE): SlackOption => {
+  const text = {
+    [PodUtil.CHANNEL_TYPE.NORMAL]: 'Normal',
+    [PodUtil.CHANNEL_TYPE.DEMO]: 'Demo',
+  }[value];
+  return {
+    text: {
+      type: 'plain_text',
+      text,
+    },
+    value,
+  };
+};
+
+const pullOption: SlackOption = {
+  text: {
+    type: 'plain_text',
+    text: 'Pull',
+  },
+  value: PodUtil.ENTRYPOINT_TYPE.PULL,
+};
+
+const pushOption: SlackOption = {
+  text: {
+    type: 'plain_text',
+    text: 'Push',
+  },
+  value: PodUtil.ENTRYPOINT_TYPE.PUSH,
+};
+
+export function getOpenCloseModal({
+  stateOrRegionName: selectedStateOrRegionName,
+  channelType: selectedChannelType,
+  channelRows = [],
+}: OpenCloseModalProps = {}): SlackView {
+  logger.info('ENTERING SLACKBLOCKUTIL.getOpenCloseModal');
+
+  // Create rows for each channel
+  const rows = channelRows.map(
+    ({ id, channelName, stateOrRegionName, entrypoints }): SlackBlock => {
+      const blockId = PodUtil.getBlockId(stateOrRegionName, channelName);
+      const text = id
+        ? `*${channelName}*`
+        : `*${channelName}* :warning: _channel not found_`;
+      const options: SlackOption[] = [pullOption, pushOption];
+      const initialOptions = entrypoints.map((type) => {
+        switch (type) {
+          case PodUtil.ENTRYPOINT_TYPE.PULL:
+            return pullOption;
+          case PodUtil.ENTRYPOINT_TYPE.PUSH:
+            return pushOption;
+        }
+        logger.error(
+          `SLACKBLOCKUTIL.getOpenCloseModal: Invalid entrypoint type for ${channelName}: ${type}`
+        );
+      });
+
+      const accessory = {
+        type: 'checkboxes',
```

Comment:
We could also switch this to `multi_static_select` which is a little more space efficient than checkboxes, but it requires more clicks to actually enable / disable stuff.